scico/solver.py

```diff
@@ -229,25 +230,36 @@ def minimize(
         min_func = _wrap_func(func_, x0_shape, x0_dtype)
         jac = False
 
-    res = spopt.minimize(
-        min_func,
-        x0=x0,
-        args=args,
-        jac=jac,
-        method=method,
-        options=options,
+    res_dict = {}
+
+    def fun(_):
```

Comment:
Does `fun` need to have one argument for `hcb` to work? If so, `x0` could be a good choice. If not, remove `_`.

Reply:
The call parameter arg requires something to be there so I made the adjustment to use x0, thanks!